core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -335,7 +336,6 @@ class KafkaController(val config : KafkaConfig, zkUtils: ZkUtils, val brokerStat
       maybeTriggerPartitionReassignment()
       maybeTriggerPreferredReplicaElection()
       /* send partition leadership info to all live brokers */
-      sendUpdateMetadataRequest(controllerContext.liveOrShuttingDownBrokerIds.toSeq)
```

Comment:
Should we still send `UpdateMetadataRequest`s here? The requests sent in line 330 were just include the information stored in ZK before the leader elections. So it seems we still need to propagate the state change (if there is any) in `replicaStateMachine.startup()` and `partitionStateMachine.startup()`.